code/game/objects/effects/decals/cleanable/misc.dm

```diff
@@ -509,6 +514,33 @@
 		ignite()
 	return ..()
 
+/obj/effect/decal/cleanable/fuel_pool/on_entered(datum/source, atom/movable/enflammable_atom)
+	. = ..()
+	if(enflammable_atom.throwing) // don't light from things being thrown over us
+		return
+	if(isitem(enflammable_atom))
+		var/obj/item/enflamed_item = enflammable_atom
+		if(enflamed_item.get_temperature() > FIRE_MINIMUM_TEMPERATURE_TO_EXIST)
+			ignite()
+	else if(isliving(enflammable_atom))
+		var/mob/living/enflamed_living = enflammable_atom
+		if(enflamed_living.on_fire)
+			ignite()
+
+/obj/effect/decal/cleanable/fuel_pool/proc/ignition_trigger(datum/source, atom/movable/enflammable_atom)
```

Comment:
As I believe I commented before this code block is identical to the one above, either make that call this proc or make both call a third proc. Don't copy/paste code.